src/main/kotlin/org/rust/ide/inspections/RsApproxConstantInspection.kt

```diff
@@ -19,11 +22,34 @@ class RsApproxConstantInspection : RsLocalInspectionTool() {
             if (literal is RsLiteralKind.Float) {
                 val value = literal.value ?: return
                 val constant = KNOWN_CONSTS.find { it.matches(value) } ?: return
-                holder.registerProblem(o, literal.suffix ?: "f64", constant)
+                val lib = when (o.containingFile.rustFile?.getStdlibAttributes(o.containingCrate)) {
+                    Attributes.NONE -> STD
+                    Attributes.NO_STD -> CORE
+                    else -> return
+                }
+                val type = literal.suffix ?: "f64"
```

Comment:
```suggestion
                val type = when (val type = o.type) {
                    is TyFloat -> type.name
                    else -> "f64"
                }
```
You can also add a test with `let a: f32 = 3.1415;` to test such inferred type case.